woocommerce/wc-api-dev
# Add Auto Update

This PR makes it so that the `wc-api-dev` plugin can auto update itself as new (non prerelease) releases come out on GitHub.

This can be disabled by setting the constant `WC_API_DEV_AUTO_UPDATE` to false.

To Test:
* This must be tested on install that does not use version control (including root WP). Optionally, you could comment out the [is_vcs_checkout check from here](https://github.com/WordPress/WordPress/blob/33047ba14e111d2935ccc5ff412e5dafaa9777cb/wp-admin/includes/class-wp-automatic-updater.php#L148).
* Install this branch into `wc-api-dev`. Download and install from zip.
* Use this [helper gist](https://gist.github.com/justinshreve/33ce0b15435a56b2c9cbdae46244b1e7). Save it as `auto-update.php` in your root WordPress folder. This is so you can force an auto update, and ignore the lock mechanism... otherwise you have to wait for transients to expire.
* Open `wc-api-dev.php` and `readme.txt` and just set the version to something lower, like 0.4.0. This is to trick the system into thinking the plugin is out of date.
* Go to the plugins page and click 'View Version 0.5.0' and you should see some basic plugin info.
* Run `php auto-update.php` from the command line in your root WordPress folder.
* WordPress should auto update. If it worked, you should see your `wc-api-dev.php` and `readme.txt` have the latest version tag, and the auto upgrade code will be gone too, since the current release doesn't have it.
